pipeline.py

```diff
@@ -72,31 +89,109 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = CorpusManager.get_articles(self.corpus_manager).values()
+
+        for article in articles:
+            raw_text = Article.get_raw_text(article)
+            tokens = self._process(raw_text)
+
+            cleaned = []
+            single_tagged = []
+            multiple_tagged = []
+            for token in tokens:
+                cleaned.append(token.get_cleaned())
+                single_tagged.append(token.get_single_tagged())
+                multiple_tagged.append(token.get_multiple_tagged())
+
+            article.save_as(' '.join(cleaned), ArtifactType.cleaned)
+            article.save_as(' '.join(single_tagged), ArtifactType.single_tagged)
+            article.save_as(' '.join(multiple_tagged), ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        clean_text = ''
+
+        for symbol in raw_text.replace('-\n', ''):
+            if symbol.isalpha():
+                clean_text += symbol
+            if symbol.isspace() or symbol == '\n':
+                clean_text += ' '
+
+        mystem = Mystem()
+        morph_analyzer = pymorphy2.MorphAnalyzer()
+
+        clean_text_analysis = mystem.analyze(clean_text)
+
+        tokens = []
+        for single_word_analysis in clean_text_analysis:
+            if single_word_analysis['text'].isalpha() and single_word_analysis['analysis']:
+                token = MorphologicalToken(single_word_analysis['text'])
+                tokens.append(token)
+
+                analysis = single_word_analysis['analysis']
+                token.normalized_form = analysis[0]['lex']
+                token.tags_mystem = analysis[0]['gr']
+
+                parses = morph_analyzer.parse(single_word_analysis['text'])
+                token.tags_pymorphy = parses[0].tag
+
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    if isinstance(path_to_validate, str):
+        path_to_validate = Path(path_to_validate)
+
+    raw_txt = 0
```

Comment:
initialize variables close to where you are using them

Reply:
Done